plugins/pur/core/src/main/java/com/pentaho/repository/importexport/StreamToTransNodeConverter.java

```diff
@@ -124,15 +133,35 @@ public IRepositoryFileData convert( final InputStream inputStream, final String
       Repository repository = connectToRepository();
       Document doc = PDIImportUtil.loadXMLFrom( inputStream );
       transMeta.loadXML( doc.getDocumentElement(), repository, false );
+
+      if ( transMeta.hasMissingPlugins() ) {
+        KettleMissingPluginsException
+          missingPluginsException =
+          new KettleMissingPluginsException( getErrorMessage( transMeta.getMissingTrans() ) );
+        throw new ConverterException( missingPluginsException );
+      }
+
       TransDelegate delegate = new TransDelegate( repository, this.unifiedRepository );
       saveSharedObjects( repository, transMeta );
       return new NodeRepositoryFileData( delegate.elementToDataNode( transMeta ), size );
-    } catch ( Exception e ) {
+    } catch ( IOException | KettleException e ) {
       logger.error( e );
```

Comment:
At least is logged here.

Reply:
I am not catching ConvertException anywhere! :P